### PR TITLE
Reduce code visibility and include only what you need from config lib…

### PR DESCRIPTION
**Pull request by baldersheim**
…rary.

@toregge or @vekterli PR

### CODE REVIEW

File: vsm/src/vespa/vsm/vsm/vsm-adapter.h [hunk 1]
```diff
@@ -97,17 +97,12 @@ typedef std::shared_ptr<DocsumTools> DocsumToolsPtr;
 class VSMConfigSnapshot {
 private:
     const vespalib::string _configId;
-    const config::ConfigSnapshot _snapshot;
+    std::unique_ptr<config::ConfigSnapshot> _snapshot;
```
Comment (vekterli) [inline vsm/src/vespa/vsm/vsm/vsm-adapter.h:100]: Should this be `unique_ptr<const config::ConfigSnapshot>` to match the previous constness?

**Review by vekterli — APPROVED** [review]
:+1: just some very minor stuff for consideration

File: config/src/vespa/config/common/configvalue.h [hunk 2]
```diff
@@ -20,10 +17,12 @@ typedef std::shared_ptr<const protocol::Payload> PayloadPtr;
  */
 class ConfigValue {
 public:
-    typedef std::unique_ptr<ConfigValue> UP;
-    ConfigValue(const std::vector<vespalib::string> & lines, const vespalib::string & xxhash);
+    ConfigValue(StringVector lines);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config/src/vespa/config/common/configvalue.h]: Fixed

File: config/src/vespa/config/common/types.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+typedef std::vector<double> DoubleVector;
+typedef std::vector<int64_t> LongVector;
+typedef std::vector<int32_t> IntVector;
+typedef std::map<vespalib::string, vespalib::string> StringMap;
+typedef std::map<vespalib::string, bool> BoolMap;
+typedef std::map<vespalib::string, double> DoubleMap;
+typedef std::map<vespalib::string, int64_t> LongMap;
+typedef std::map<vespalib::string, int32_t> IntMap;
+
+} // namespace common
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config/src/vespa/config/common/types.h]: Removed

File: config/src/vespa/config/set/configsetsourcefactory.h [hunk 4]
```diff
@@ -16,14 +16,10 @@ class ConfigKey;
 class ConfigSetSourceFactory : public SourceFactory
 {
 public:
-    typedef ConfigSetSource::BuilderMap BuilderMap;
-    typedef ConfigSetSource::BuilderMapSP BuilderMapSP;
-    ConfigSetSourceFactory(const BuilderMapSP & builderMap);
+    using BuilderMapSP = ConfigSetSource::BuilderMapSP;
+    ConfigSetSourceFactory(BuilderMapSP builderMap);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config/src/vespa/config/set/configsetsourcefactory.h]: Fixed

File: config/src/vespa/config/subscription/confighandle.h [hunk 5]
```diff
@@ -16,7 +17,7 @@ class ConfigHandle
 public:
     typedef std::unique_ptr<ConfigHandle <ConfigType> > UP;
 
-    ConfigHandle(const ConfigSubscription::SP & subscription);
+    ConfigHandle(std::shared_ptr<ConfigSubscription> subscription);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config/src/vespa/config/subscription/confighandle.h]: Fixed

File: config/src/vespa/config/subscription/configsubscriptionset.h [hunk 6]
```diff
@@ -25,7 +27,7 @@ class ConfigSubscriptionSet
      *
      * @param context A ConfigContext shared between all subscriptions.
      */
-    ConfigSubscriptionSet(const IConfigContext::SP & context);
+    ConfigSubscriptionSet(std::shared_ptr<IConfigContext> context);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config/src/vespa/config/subscription/configsubscriptionset.h]: Fixed

File: storageserver/src/vespa/storageserver/app/process.h [hunk 7]
```diff
@@ -42,7 +42,7 @@ class Process : public ApplicationGenerationFetcher {
     using milliseconds = std::chrono::milliseconds;
 
     Process(const config::ConfigUri & configUri);
-    virtual ~Process() {}
+    virtual ~Process();
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply storageserver/src/vespa/storageserver/app/process.h]: Fixed

File: vsm/src/vespa/vsm/vsm/vsm-adapter.h (hunk 1 above)
Comment (baldersheim) [reply vsm/src/vespa/vsm/vsm/vsm-adapter.h]: Fixed

File: config/src/vespa/config/subscription/configsubscriber.h [hunk 8]
```diff
@@ -45,7 +45,7 @@ class ConfigSubscriber
      *
      * @param context A ConfigContext shared between all subscribers.
      */
-    ConfigSubscriber(const IConfigContext::SP & context);
+    ConfigSubscriber(std::shared_ptr<IConfigContext> context);
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply config/src/vespa/config/subscription/configsubscriber.h]: Fixed

**Pull request merged**